src/opendap/coreServlet/BotBlocker.java

```diff
@@ -145,40 +145,38 @@ public boolean requestCanBeHandled(HttpServletRequest request)
             throws Exception {
 
         String remoteAddr = request.getRemoteAddr();
-
         if(ipAddresses.contains(remoteAddr)){
             log.info("The ip address: {} is " +
                     "on the list of blocked addresses", LogUtil.scrubEntry(remoteAddr));
-
-            if(responseFiltering)
+            if(responseFiltering) {
```

Comment:
So if responseFiltering is false, and remoteAddr is on/in ipAddresses, then requestCanBeHandled() is true even if the address is on the list, but if responseFiltering is true, then requestCanBeHandled() will be false if isResponseBlocked() is false.

I'm confused. It seems like the logic is reversed.

Reply:
The `BotBlocker` "claims" the request when the request should be blocked (`BotBlocker.requestCanBeHandled()` returns true). After the request is "claimed" the dispatcher calls the `BotBlocker.handleRequest()` method which returns a Forbidden error to the client.